--- ../luasrc/dispatcher.lua	2022-04-24 02:52:01.345276400 +0800
+++ ../luasrc/dispatcher.lua	2022-04-24 02:55:41.187918800 +0800
@@ -461,7 +461,7 @@
 	context.request = r
 
 	local fs = require "nixio.fs"
-	local default_path_info = fs.access("/etc/config/wizard") and "admin/initsetup" or ""
+	local default_path_info = fs.access("/etc/config/wizard") and "admin/system/initsetup" or ""
 
 	local pathinfo = http.urldecode(request:getenv("PATH_INFO") or default_path_info, true)
 
@@ -928,9 +928,14 @@
 				return tpl.render("sysauth", scope)
 			end
 
-			http.header("Set-Cookie", 'sysauth=%s; path=%s; SameSite=Strict; HttpOnly%s' %{
-				sid, build_url(), http.getenv("HTTPS") == "on" and "; secure" or ""
-			})
+			local cookie_p = uci:get("wizard", "default", "cookie_p")
+			local timeout = 'Thu, 01 Jan 3000 01:00:00 GMT'
+			if cookie_p == '0' then
+				timeout = ''
+			end
+			http.header("Set-Cookie", 'sysauth=%s; expires=%s; path=%s; SameSite=Strict; HttpOnly%s' %{
+				sid, timeout, build_url(), http.getenv("HTTPS") == "on" and "; secure" or ""
+ 			})
 
 			local lang = http.formvalue("luci_lang")
 			if lang then
@@ -961,8 +966,14 @@
 	if #lookup_ctx.acls > 0 then
 		local perm = check_acl_depends(lookup_ctx.acls, ctx.authacl and ctx.authacl["access-group"])
 		if perm == nil then
-			http.status(403, "Forbidden")
-			return
+			local sid = context.authsession
+			if sid then
+				util.ubus("session", "destroy", { ubus_rpc_session = sid })
+				luci.http.header("Set-Cookie", "sysauth=%s; expires=%s; path=%s" %{
+					'', 'Thu, 01 Jan 1970 01:00:00 GMT', build_url()
+				})
+			end
+			luci.http.redirect(build_url())
 		end
 
 		if page then
